libevmasm/Assembly.h

```diff
@@ -162,6 +168,22 @@ class Assembly
 
 	bool isCreation() const { return m_creation; }
 
+	/// Set the source name list.
+	void setSources(std::vector<std::shared_ptr<std::string const>> _sources)
+	{
+		m_sources = std::move(_sources);
+	}
+
+	/// Set the source name list from simple vector<string>.
+	void setSources(std::vector<std::string> const& _sources)
+	{
+		for (auto const& item: _sources)
+			m_sources.emplace_back(std::make_shared<std::string>(item));
+	}
+
+	/// @returns List of source names.
+	std::vector<std::shared_ptr<std::string const>> sources() const& { return m_sources; }
```

Comment:
Oh, I think that was a typo.

Reply:
I have another question - why a `vector` of `shared_ptr` of `string`s instead of just a `vector` of `string`s?

Reply:
I actually asked the same thing in one of the other comments :) I think it's just a micro optimization to use less memory (we do things like that e.g. to reuse source locations). I'm not sure it's particularly necessary though. I suspect this might have been based on something that's already in the code and does this.

Reply:
Returning a const reference to a vector of strings is most definitely cheaper than returning a copy of a vector of shared_ptrs.

Reply:
Changed to `std::vector<std::string> const& sourceUnitNames()`.